src/wallet/wallet.ts

```diff
@@ -2430,11 +2452,29 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     );
 
     // Create a wrapper around this wallet to override getFullTxById
-    const wrappedWallet = {
-      ...this,
-      getFullTxById: this.getFullTxByIdForNanoContract.bind(this),
-      getNetworkObject: this.getNetworkObject.bind(this),
-    };
+    const wrappedWallet = new Proxy(this, {
```

Comment:
question(non-blocking): This is a wild solution! It works, but it can get hard to maintain properly in the future. The jump from a 5-line simple-object wrapper to a 20+ line dynamically-binded one in this PR is evidence of that.

Could we create an issue to rethink this solution in the future?